app/lib/task/global_lock.dart

```diff
@@ -63,21 +63,19 @@ class GlobalLock {
           // Try to refresh, if claim is still valid and we're not done.
           if (c.valid && !done.isCompleted) {
             refreshed = c.refresh();
-            try {
-              if (!await refreshed) {
-                _log.warning(
-                  'failed to refresh claim $claimId on $_lockId',
-                );
-                return; // Stop refreshing, if refresh failed.
-              }
-            } on Exception catch (e, st) {
-              // Stop refreshing, and log the error
-              _log.severe('Error refreshing claim $claimId on $_lockId', e, st);
-              return;
+            if (!await refreshed) {
+              _log.warning('failed to refresh claim $claimId on $_lockId');
+              return; // Stop refreshing, if refresh failed.
             }
           }
         }
-      });
+      } on Exception catch (e, st) {
+        // Stop refreshing, and log the error
+        _log.severe('Error refreshing claim $claimId on $_lockId', e, st);
```

Comment:
If we want to harden the micro-task for unhandled exceptions/errors.

I'd suggest keeping the try/catch we had before and wrap everything inside `scheduleMicrotask` in a new one that has a more correct error message.

This says that there was an error during refreshing. But that's only the case if it was thrown by `await refreshed`.

Thus, I'd suggest that a try-catch that covers the entire body of the microtask should say something less clear. Like `Internal error while maintaining claim for GlobalLock`. And it should probably catch both errors and exceptions. To be clear such a high-level try catch is not there because there are any errors/exceptions to catch -- I don't see any. But it's there because: if there is an internal error, then we can stop refreshing the claim, let it expire and continue gracefully -- we do not need to crash the whole application. Ofcourse this is a risk, because we don't know why this internal error happened, whether it's temporary or permanent and whether it could be resolved by restarting.